### PR TITLE
core: merge kernel_macros.h and attributes.h into kernel_defines.h

### LINKED ISSUE #1739: Merge kernel_macros.h and attributes.h?
Splitting this in two categories is quite arbitrary, isn't it?

I like the name "attributes.h" better, but it is likely to clash with application header files.

### PR DESCRIPTION
**Pull request by OlegHahm**
Fixes #1739.

### CODE REVIEW

File: core/include/kernel_defines.h [hunk 1]
```diff
@@ -26,6 +26,39 @@
...
+        (__extension__ ({ \
+            __extension__ const __typeof__ (((TYPE *) 0)->MEMBER) *__m____ = (PTR); \
+            ((TYPE *) ((char *) __m____ - offsetof(TYPE, MEMBER))); \
+        }))
+#else
+#   define container_of(PTR, TYPE, MEMBER) \
+        ((TYPE *) ((char *) (PTR) - offsetof(TYPE, MEMBER)))
+#endif
+
+
```
Comment (gebart) [inline core/include/kernel_defines.h]: remove this extra newline

**Comment by gebart** [comment]
First ACK, needs a second one

File: core/include/kernel_defines.h (hunk 1 above)
Comment (OlegHahm) [inline core/include/kernel_defines.h]: addressed and squashed immediately.

**Comment by OlegHahm** [comment]
@Kijewski, @kaspar030, anyone else care to give a second ACK?

**Comment by LudwigKnuepfer** [comment]
It says:
```
 #define ATTRIBUTES_H_
```

**Comment by OlegHahm** [comment]
Fixed and squashed immediately.

**Comment by OlegHahm** [comment]
@Kijewski, @kaspar030, @Kijewski, @haukepetersen, anyone cares for a second ACK?

**Comment by Kijewski** [comment]
ACK

**Pull request merged**